src/Symfony/Bridge/Twig/Translation/TwigExtractor.php

```diff
@@ -52,16 +52,22 @@ public function __construct(\Twig_Environment $twig)
     /**
      * {@inheritdoc}
      */
-    public function extract($directory, MessageCatalogue $catalogue)
+    public function extract($resource, MessageCatalogue $catalogue)
     {
-        // load any existing translation files
-        $finder = new Finder();
-        $files = $finder->files()->name('*.twig')->sortByName()->in($directory);
+        if (is_array($resource)) {
+            $files = array_filter($resource, function ($filename) {
```

Comment:
Comments addressed. Also added more tests and different options for `$resource` like support for `Traversable`, a string containing a file, etc. Updated interface documentation too.